pkg/bpf-mgr/lib-bpfman.go

```diff
@@ -63,6 +63,8 @@ func bpfmanCreateNodeFirewallApplication(ctx context.Context, c client.Client, o
 			return fmt.Errorf("failed to get BpfApplication: %v for obj: %s", err, obj.Name)
 		}
 	} else {
+		// object exists repopulate it with the new configuration and update it
+		prepareBpfApplication(&bpfApp, obj, dbg)
```

Comment:
Should you move the prepareBpfApplication() call on line 53 to after line 57?  As-is, I think in the case where the object is found, the bpfApp created on line 53 is being overritten by the c.Get() call.

Reply:
if Get fail it won't be overwritten I think, but maybe saver move inside

Reply:
I agree.  I think it works like it is, but you're calling it twice for the case where the get succeeds and I don't think you need to.

Reply:
done